corehq/apps/locations/models.py

```diff
@@ -282,7 +282,7 @@ def couch_locations(self, wrapped=True):
 
     def accessible_to_user(self, domain, user):
         if user.has_permission(domain, 'access_all_locations'):
-            return self.all()
+            return self.filter(domain=domain)
```

Comment:
@esoergel just thinking if the same should be done at https://github.com/dimagi/commcare-hq/pull/14255/files#diff-a2c9183455842303ef415bfcd036fdc3R290 for uniformity.